### PR TITLE
chore(sweeps): set sweep state on scheduler exit

### PR DESCRIPTION
**Pull request by gtarpenning**
Fixes
-----

https://wandb.atlassian.net/browse/WB-13456

Screen Shot 2023-05-31 at 5 31 53 PM

Description
-----------
Sweeps that end without the agent hitting the AgentHeartbeat path in gorilla end up stuck in "running" or starting state, this manually sets the sweep state to FINISHED when the sweep exits.

### CODE REVIEW

File: wandb/sdk/launch/sweeps/scheduler.py [hunk 1]
```diff
@@ -550,6 +551,13 @@ def _create_run(self) -> Dict[str, Any]:
             )
         return {}
 
+    def _set_sweep_state(self, state: str) -> None:
```
Comment (KyleGoyette) [inline wandb/sdk/launch/sweeps/scheduler.py:554]: This seems like it should also be used elsewhere? Will there be a follow up with other uses for this?

**Review by KyleGoyette — APPROVED** [review]

File: wandb/sdk/launch/sweeps/scheduler.py (hunk 1 above)
Comment (gtarpenning) [reply wandb/sdk/launch/sweeps/scheduler.py:554]: This only happens when the sweep scheduler ends when it hits a run_cap (previously managed by gorilla, but now handled by the scheduler). We then don't mark the sweep as finished in the backend. I think this is the only case where this happens...

Comment (gtarpenning) [reply wandb/sdk/launch/sweeps/scheduler.py:556]: JK I guess we also have to call this when the scheduler starts...

**Pull request merged**